universal-application-tool-0.0.1/test/services/applicant/question/EnumeratorQuestionTest.java

```diff
@@ -0,0 +1,110 @@
+import services.question.types.RepeaterQuestionDefinition;
+import support.QuestionAnswerer;
+import support.TestQuestionBank;
+
+@RunWith(JUnitParamsRunner.class)
+public class EnumeratorQuestionTest {
+  private static final RepeaterQuestionDefinition enumeratorQuestionDefinition =
+      new RepeaterQuestionDefinition(
+          "household members",
+          Path.create("applicant.household_members[]"),
+          Optional.empty(),
+          "description",
+          ImmutableMap.of(Locale.US, "question?"),
+          ImmutableMap.of(Locale.US, "help text"));
+
+  private Applicant applicant;
+  private ApplicantData applicantData;
+
+  private static final TestQuestionBank testQuestionBank = new TestQuestionBank(false);
+
+  @Before
+  public void setUp() {
+    applicant = new Applicant();
+    applicantData = applicant.getApplicantData();
+    testQuestionBank.reset();
+  }
+
+  @Test
+  public void withEmptyApplicantData() {
+    ApplicantQuestion applicantQuestion =
+        new ApplicantQuestion(
+            enumeratorQuestionDefinition, applicantData, ApplicantData.APPLICANT_PATH);
+
+    RepeaterQuestion enumeratorQuestion = new RepeaterQuestion(applicantQuestion);
+
+    assertThat(enumeratorQuestion.isAnswered()).isFalse();
+    assertThat(enumeratorQuestion.hasTypeSpecificErrors()).isFalse();
+    assertThat(enumeratorQuestion.hasQuestionErrors()).isFalse();
+  }
+
+  @Test
+  public void withApplicantData_passesValidation() {
+    ApplicantQuestion applicantQuestion =
+        new ApplicantQuestion(
+            enumeratorQuestionDefinition, applicantData, ApplicantData.APPLICANT_PATH);
+    QuestionAnswerer.answerEnumeratorQuestion(
+        applicantData,
+        applicantQuestion.getContextualizedPath(),
+        ImmutableList.of("first", "second", "third"));
+
+    RepeaterQuestion enumeratorQuestion = new RepeaterQuestion(applicantQuestion);
+
+    assertThat(enumeratorQuestion.isAnswered()).isTrue();
+    assertThat(enumeratorQuestion.getEntityNames()).contains("first", "second", "third");
+    assertThat(enumeratorQuestion.hasTypeSpecificErrors()).isFalse();
+    assertThat(enumeratorQuestion.hasQuestionErrors()).isFalse();
+  }
+
+  @Test
+  @Parameters({"", " "})
+  public void withBlankStrings_hasValidationErrors(String value) {
+    ApplicantQuestion applicantQuestion =
+        new ApplicantQuestion(
+            enumeratorQuestionDefinition, applicantData, ApplicantData.APPLICANT_PATH);
+    QuestionAnswerer.answerEnumeratorQuestion(
+        applicantData, applicantQuestion.getContextualizedPath(), ImmutableList.of(value));
+
+    RepeaterQuestion enumeratorQuestion = new RepeaterQuestion(applicantQuestion);
+
+    assertThat(enumeratorQuestion.isAnswered()).isTrue();
+    assertThat(enumeratorQuestion.getEntityNames()).contains(value);
+    assertThat(enumeratorQuestion.hasTypeSpecificErrors()).isFalse();
+    assertThat(enumeratorQuestion.hasQuestionErrors()).isTrue();
```

Comment:
nit: consider checking the error message, too

Reply:
done